dashboard/app/models/user.rb

```diff
@@ -1252,6 +1252,26 @@ def can_edit_password?
     encrypted_password.present?
   end
 
+  def can_edit_display_name?
+    !picture_or_word_account?
+  end
+
+  def picture_or_word_account?
+    return false if sections_as_student.empty?
+    return false if encrypted_password.present?
+    sections_as_student.all? do |section|
+      section.login_type == Section::LOGIN_TYPE_PICTURE || section.login_type == Section::LOGIN_TYPE_WORD
+    end
```

Comment:
OK, I'll dig into this a little bit more, and try to think about what the correct behavior actually should be in those cases, as well.